pkg/resources/konnectivity/sidecar.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package konnectivity
+
+import (
+	"fmt"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// ProxySidecar returns container that runs konnectivity proxy server as a sidecar in apiserver pods.
+func ProxySidecar(serverCount int32) (*corev1.Container, error) {
+	return &corev1.Container{
+		Name:            "konnectivity-server-container",
+		Image:           "us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-server:v0.0.24",
```

Comment:
(in whole PR) do not hardcode the image registry, use the [ImageRegistry](https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/data.go#L403-L409) helper to allow overring it.

Reply:
User-cluster-controller doesn't seem to have way of overriding registry. I have hard coded it there right now.

Reply:
Seed cluster controller is fixed. Pushing in a moment.